tests/e2e/e2e_setup_test.go

```diff
@@ -391,84 +387,116 @@ func noRestart(config *docker.HostConfig) {
 	}
 }
 
-func (s *IntegrationTestSuite) initUpgrade() {
+func (s *IntegrationTestSuite) upgrade() {
 	// submit, deposit, and vote for upgrade proposal
-	s.submitProposal(s.chains[0])
-	s.submitProposal(s.chains[1])
+	currentHeightA := s.chainHeight(s.valResources[s.chains[0].ChainMeta.Id][0].Container.ID)
+	currentHeightB := s.chainHeight(s.valResources[s.chains[1].ChainMeta.Id][0].Container.ID)
+	propHeightA = currentHeightA + 40 + len(s.chains[0].Validators)
+	propHeightB = currentHeightB + 40 + len(s.chains[1].Validators)
+
+	s.submitProposal(s.chains[0], propHeightA)
+	s.submitProposal(s.chains[1], propHeightB)
 	s.depositProposal(s.chains[0])
 	s.depositProposal(s.chains[1])
-	s.voteProposal(s.chains[0])
-	s.voteProposal(s.chains[1])
-
+	var wg sync.WaitGroup
+	wg.Add(2)
+	// use goroutines to vote on both chains concurrently (no possibility of sequence mismatch)
+	go s.voteProposal(s.chains[0], &wg)
+	go s.voteProposal(s.chains[1], &wg)
```

Comment:
Since these are 2 independent chains, why does it matter if we are voting concurrently?

Reply:
It's more of an opportunity than a need. If we have 100 validators on each chain, instead of looping through 200 validators we instead can do them at the same time since there is no risk of account sequence mismatches.

Reply:
Do we have plans to run that many, and would the CI even handle it?

The reason I'm careful about concurrency in CI is that it always adds a layer of complexity. Since we are not unit testing this setup code directly, it's easy to miss something and struggle in the future with "unintended consequences".

Imo CI should be as simple as possible and things like concurrencyy added when there is a concrete need for it.

I don't have a strong opinion on this. Up to you to decide. Just sharing my thoughts :+1:

Reply:
If voting doesn't take that much time (e.g. <1m) we should probably just make it sequential.

Reply:
Okay, I reverted it back to sequential. The only downside is that VotingPeriod for B now has to be the originally calculated VotingPeriod for B plus VotingPeriod for A since A votes first now, but its not a huge deal.